lib/matplotlib/pyplot.py

```diff
@@ -1228,11 +1249,29 @@ def subplot(*args, **kwargs):
         (ax for ax in fig.axes
          if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key),
         None)
+    # if we found an axes at the right position sort out if we
+    # should re-use it
+    if ax is not None:
+        # sort out what projection and class would be created so we
+        # can tell if we should re-use an existing one
+        p_class, pkw = fig._process_projection_requirements(*args, **kwargs)
+        # if no projection or kwargs were passed, return the axes we
+        # found.
+        if projection is proj_placeholder and pkw == {}:
```

Comment:
shouldn't this case be handled by the more general case below with `_projection_init == (Axes, {})`?

Reply:
No, because we did not used to include the projection class in the key, only the kwargs, for the matching, and then did isintance checking to see if the Axes was a sub-class of the Axes we found.  The first condition is for the "just get me that Axes" logic, the second is for if the user passed anything that we need to resolve.